src/main/scala/com/kinja/amqp/persistence/MySqlMessageStore.scala

```diff
@@ -60,16 +61,57 @@ abstract class MySqlMessageStore extends MessageStore {
 		def autoInc =
 			id.? ~ channelId ~ deliveryTag ~ multiple ~ createdTime <> (MessageConfirmation.apply _, MessageConfirmation.unapply _) returning id
 	}
+
 	private object Queries {
+		def deleteMessagesWithMatchingMultiConfirms() =
+			sqlu"""
+				DELETE
+					m.*
+				FROM
+					rabbit_messages m
+				JOIN
+					rabbit_confirmations c
+						ON m.channelId = c.channelId
+						AND m.deliveryTag <= c.deliveryTag
+				WHERE
+					c.multiple = 1
+			"""
 
-		def commit() =
+		def deleteSingleConfIfNoMatchingMsg(olderThan: String) =
```

Comment:
As the name suggests, I think we should do a confirmation-message join here as well.